crates/primitives/src/prune/mod.rs

```diff
@@ -91,21 +91,76 @@ impl ReceiptsLogPruneConfig {
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
 pub enum PruneProgress {
     /// There is more data to prune.
-    HasMoreData,
+    HasMoreData(PruneInterruptReason),
     /// Pruning has been finished.
     Finished,
 }
 
+/// Reason for interrupting a prune job.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum PruneInterruptReason {
+    /// Prune job timed out.
+    Timeout,
+    /// Limit on the number of deleted entries (rows in the database) per prune job was reached.
+    LimitEntriesDeleted,
```

Comment:
maybe something like that?
```suggestion
    DeletedEntriesLimitReached,
```